libs/books/data-access/src/lib/+state/reading-list.reducer.ts

```diff
@@ -58,6 +58,14 @@ const readingListReducer = createReducer(
   ),
   on(ReadingListActions.failedRemoveFromReadingList, (state, action) =>
     readingListAdapter.addOne(action.item, state)
+  ),
+  on(ReadingListActions.markAsFinished, (state, action) =>
+    readingListAdapter.updateOne({id: action.item.bookId, changes: 
+      { finished: true, finishedDate: new Date().toISOString() }}, state)
```

Comment:
You don't need to change it, but reducers should _always_ be pure. In this case the `finishedDate` should be a property of the `markAsFinished` action.